### PR TITLE
updated HEKClient to return json again

### LINKED ISSUE #3898: SunPy’s HEK module
I'm new in Sunpy and I am tring to learn how to use the HEK module in order to obtain a list of events (for example FL ).

When I compile I obtain 
"'utf-8' codec can't decode byte 0xc5 in position 10559: invalid continuation byte",

The code used is
1. from sunpy.net import hek
2. client = hek.HEKClient()  
3. tstart = '2011/08/09 07:23:56'
4.tend = '2011/08/09 12:40:29'
5.event_type = 'FL'
6.result = client.search(hek.attrs.Time(tstart,tend),hek.attrs.EventType(event_type)) 

System details are:

  - SunPy Version: 1.1.1
  - Astropy Version: 4.0
  - Python Version: 3.7.6
  - OS information: Windows 10 10.0.18362 (Intel64 Family 6 Model 58 Stepping 9, GenuineIntel)

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes #3898

### CODE REVIEW

File: changelog/3899.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+`sunpy.net.hek.HEKClient` was returning HTML and not JSON.
```
Comment (abhijeetmanhas) [inline changelog/3899.bugfix.rst]: Should it be `~sunpy.net.hek.HEKClient` ?

Comment (nabobalis) [reply changelog/3899.bugfix.rst]: It can be.

Comment (nabobalis) [inline changelog/3899.bugfix.rst]:
```suggestion
`~sunpy.net.hek.HEKClient` was returning HTML and not JSON.
```

**Comment by nabobalis** [comment]
Azure is not running for some reason.

File: sunpy/net/hek/hek.py [hunk 2]
```diff
@@ -61,14 +54,14 @@ def _download(self, data):
...
         while True:
             data['page'] = page
-            fd = urllib.request.urlopen(
-                self.url, urllib.parse.urlencode(data).encode('utf-8'))
+            fd = urllib.request.urlopen(self.url+urllib.parse.urlencode(data))
             try:
-                result = json.load(reader(fd))
+                result = json.load(fd)
+            except Exception as e:
+                raise IOError(f"Failed to load return from the HEKClient due to {e}")
```
Comment (Cadair) [inline sunpy/net/hek/hek.py]:
```suggestion
                raise IOError(f"Failed to load return from the HEKClient.") from e
```

**Comment by fluxtransport** [comment]
Hi @nabobalis and @Cadair,
We just had someone report that HEK queries were not working. It had to do with us the server config's changing. The DEFAULT_URL needs to be changed from http -> https. I was going to do a pull request but then saw you're updating the HEKClient. So thought I should mention it here first.

**Comment by nabobalis** [comment]
I didn't update to https but I should do that as well. 

Thanks for telling us!

**Comment by fluxtransport** [comment]
Thanks!

**Pull request merged**